compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertRegionToWorkgroups.h

```diff
@@ -22,18 +24,30 @@ namespace Flow {
 class DispatchRegionOp;
 class DispatchWorkgroupsOp;
 
-/// A function that builds the workload body of a DispatchWorkgroupsOp.
-using WorkloadBuilderFn =
-    std::function<void(OpBuilder &, Location, ArrayRef<BlockArgument>)>;
+/// A data structure that holds workload operands and a function that build
+/// the workload region of a WorkgroupsOp.
+struct WorkloadBuilder {
+  /// A function that builds the workload region of a WorkgroupsOp.
+  using RegionBuilderFn =
+      std::function<void(OpBuilder &, Location, ArrayRef<BlockArgument>)>;
+
+  /// Values to be used as `workload` operands of a WorkgroupsOp.
```

Comment:
Does this need to be part of the struct... Maybe you can make this part of the arguments fo RegionBuilderFn`.

Edit: Then you actually dont need a struct. You just need a callback function.